AppLovin MAX Demo App - ObjC/AppLovin MAX Demo App - ObjC/MAX/Native Ads/ALMAXManualNativeLateBindingAdViewController.m

```diff
@@ -0,0 +1,142 @@
+
+@property (nonatomic, weak) IBOutlet UIView *nativeAdContainerView;
+@property (nonatomic, weak) IBOutlet UIButton *showAdButton;
+
+@property (nonatomic, strong) MANativeAdLoader *nativeAdLoader;
+@property (nonatomic, strong) MANativeAdView *nativeAdView;
+@property (nonatomic, strong, nullable) MAAd *nativeAd;
+
+@end
+
+@implementation ALMAXManualNativeLateBindingAdViewController
+
+#pragma mark - View Lifecycle
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.nativeAdLoader = [[MANativeAdLoader alloc] initWithAdUnitIdentifier: @"YOUR_AD_UNIT"];
+    self.nativeAdLoader.nativeAdDelegate = self;
+    self.nativeAdLoader.revenueDelegate = self;
+}
+
+- (void)dealloc
+{
+    [self cleanUpAdIfNeeded];
+    
+    self.nativeAdLoader.nativeAdDelegate = nil;
+    self.nativeAdLoader.revenueDelegate = nil;
+}
+
+- (void)cleanUpAdIfNeeded
+{
+    // Clean up any pre-existing native ad to prevent memory leaks
+    if ( self.nativeAd )
+    {
+        [self.nativeAdLoader destroyAd: self.nativeAd];
+    }
+    
+    if ( self.nativeAdView )
+    {
+        [self.nativeAdView removeFromSuperview];
+    }
+}
+
+- (MANativeAdView *)createNativeAdView
+{
+    UINib *nativeAdViewNib = [UINib nibWithNibName: @"NativeManualAdView" bundle: NSBundle.mainBundle];
+    MANativeAdView *nativeAdView = [nativeAdViewNib instantiateWithOwner: nil options: nil].firstObject;
+    
+    MANativeAdViewBinder *binder = [[MANativeAdViewBinder alloc] initWithBuilderBlock:^(MANativeAdViewBinderBuilder *builder) {
+        builder.titleLabelTag = 1001;
+        builder.advertiserLabelTag = 1002;
+        builder.bodyLabelTag = 1003;
+        builder.iconImageViewTag = 1004;
+        builder.optionsContentViewTag = 1005;
+        builder.mediaContentViewTag = 1006;
+        builder.callToActionButtonTag = 1007;
+    }];
+    [nativeAdView bindViewsWithAdViewBinder: binder];
+    
+    return nativeAdView;
+}
+
+#pragma mark - IB Actions
+
+- (IBAction)loadAd
+{
+    [self cleanUpAdIfNeeded];
+    
+    [self.nativeAdLoader loadAd];
+}
+
+- (IBAction)showAd
+{
+    self.nativeAdView = [self createNativeAdView];
+    [self.nativeAdLoader renderNativeAdView: self.nativeAdView withAd: self.nativeAd];
+    [self.nativeAdContainerView addSubview: self.nativeAdView];
+    
+    [self.showAdButton setEnabled: NO];
+}
+
+#pragma mark - NativeAdDelegate Protocol
+
+- (void)didLoadNativeAd:(MANativeAdView *)nativeAdView forAd:(MAAd *)ad
+{
+    [self logCallback: __PRETTY_FUNCTION__];
+    
+    // Save ad to be rendered later
+    self.nativeAd = ad;
+    
+    [self.showAdButton setEnabled: YES];
+    
+    // Set to false if modifying constraints after adding the ad view to your layout
+    self.nativeAdContainerView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    // Set ad view to span width and height of container and center the ad
+    [self.nativeAdContainerView.widthAnchor constraintEqualToAnchor: nativeAdView.widthAnchor].active = YES;
```

Comment:
We should be moving these constraints to `showAd`.

Reply:
I tried to move them to showAd method right after [self.nativeAdContainerView addSubview: self.nativeAdView]; , but it is showing like this:
![Simulator Screen Shot - iPhone 11 - 2022-03-13 at 23 53 05](https://user-images.githubusercontent.com/95258998/158120557-f46420c2-fe7d-4b09-bb58-29aa9761fa3b.png)